### PR TITLE
SDL2 Support

### PR DESCRIPTION
**Pull request by JonasGilg**
This PR adds the following things:

- A new SDL2 Windowing Toolkit
- Three drivers for an SDL2 Keyboard
  1. Raw keyboard state driver
  2. Event driven driver
  3. Complex text input driver
- A driver for an SDL2 Mouse
- A driver for an SDL2 Controller
  - The controller driver should be able to handle all controllers, that look like an XBox or Playstation controller.
  - Hot reload of controllers is supported

Currently the PR is in a robust and functional state, but some minor tasks remain:
- [x] Extensive documentation
- [x] Improve the loading of fonts and controller DB
- [x] Controller touchpad support
- [x] Controller sensor support
- [x] SDL2_ttf CMake find script isn't following Vistas pattern
- [x] Offer SDL2 OpenVR windowing toolkit?

Things that are possible, but probably go in another PR:
- Handling of multiple windows in Keyboard, Mouse and Controller drivers
- Supporting multiple controllers at the same time
- SDL2 force feedback driver
- SDL2 touch and multitouch driver
- SDL2 audio driver

### CODE REVIEW

**Comment by JonasGilg** [comment]
If you require I can add some script in the coming days guiding through a test setup.

File: VistaCoreLibs/VistaKernel/DisplayManager/SDL2WindowImp/VistaSDL2WindowingToolkit.cpp [hunk 1]
```diff
@@ -0,0 +1,1495 @@
...
+      , cursorEnabled(true)
+      , cursor(-1)
+      , isOffscreenBuffer(false)
+      , fboId(0)
+      , fboColorId(0)
+      , fboDepthId(0)
+      , fboStencilId(0)
+      , blitFboId(0)
+      , blitFboColorId(0)
+      , blitFboDepthId(0) {
```
Comment (Schneegans) [inline VistaCoreLibs/VistaKernel/DisplayManager/SDL2WindowImp/VistaSDL2WindowingToolkit.cpp:102]: Maybe these could be initialized inline?

File: VistaDemo/configfiles/display_desktop.ini [hunk 2]
```diff
@@ -5,6 +5,16 @@
 [SYSTEM]
 DISPLAYSYSTEMS 		= MAIN
 WINDOWINGTOOLKIT	= GLUT
```
Comment (Schneegans) [inline VistaDemo/configfiles/display_desktop.ini:7]: Should this default to SDL2 now?

**Review by Schneegans — CHANGES_REQUESTED** [review]
Thanks a lot! This is a really cool improvement.

**Comment by JonasGilg** [comment]
It is actually possible to allow SDL to handle controller input without window focus using the following hint: `SDL_SetHint( SDL_HINT_JOYSTICK_ALLOW_BACKGROUND_EVENTS, "1" );`

Should we enable it?

**Review by Schneegans — APPROVED** [review]

**Pull request merged**